src/ClientLibrary/FactoryOrchestratorClient.cs

```diff
@@ -499,6 +521,11 @@ private Exception CreateIpcException(Exception ex)
                 IsConnected = false;
                 ex = new FactoryOrchestratorConnectionException(IpAddress);
             }
+            else if (ex is AuthenticationException)
+            {
+                IsConnected = false;
+                ex = new FactoryOrchestratorConnectionException(ex.Message);

```

Comment:
write a new error string informing the user the cert failed validation (you cant say hash mismatch cuz they might have provided a custom callback). Then pass in the inner exception too: https://github.com/microsoft/FactoryOrchestrator/blob/c8c8cd5733eda589b423efc8161758b4c321950b/src/ClientLibrary/FactoryOrchestratorClient.cs#L676